torch/_inductor/ir.py

```diff
@@ -5787,12 +5790,15 @@ def __init__(self, layout, inputs, constant_args, reduce_op):
         super().__init__(layout, inputs, constant_args)
         self.reduce_op = reduce_op
 
+    def get_mutation_names(self):
```

Comment:
would it make sense to apply these changes to InPlaceCollectiveKernel instead of the 2 allreduce variants?

Also, iirc Inductor's AllReduce isn't _always_ allowed to reuse its input buffer.  In cases that Inductor didn't allow reuse, should we report no aliasing?

Reply:
yeah I thought about it and I think it would be nice to apply this change only to InPlaceCollectiveKernel, but at that level we also don't know which inputs to the kernel is in-place vs not (maybe I am thinking too much ahead but there could be kernels taking two inputs, one in-place and one not). Annotating at the actual collective IR level will ensure we have all the information I believe.

Reply:
ok. well, its worth asking what the point of that base class is. does it already make any such assumptions about inplace inputs? anyway, its more of a nit, up to you